tsl/org.testeditor.tsl.dsl.web/src/org/testeditor/tsl/dsl/web/TslWebSetup.xtend

```diff
@@ -0,0 +1,36 @@
+/*
+ * generated by Xtext 2.9.0
+ */
+package org.testeditor.tsl.dsl.web
+
+import com.google.inject.Guice
+import com.google.inject.Injector
+import com.google.inject.Provider
+import com.google.inject.util.Modules
+import java.util.concurrent.ExecutorService
+import org.eclipse.xtend.lib.annotations.FinalFieldsConstructor
+import org.testeditor.tsl.dsl.TslRuntimeModule
+import org.testeditor.tsl.dsl.TslStandaloneSetup
+import org.eclipse.emf.ecore.EPackage
+import org.testeditor.tsl.TslPackage
+
+/**
+ * Initialization support for running Xtext languages in web applications.
+ */
+@FinalFieldsConstructor
+class TslWebSetup extends TslStandaloneSetup {
+	
+	val Provider<ExecutorService> executorServiceProvider;
+	
+	override Injector createInjector() {
+		val runtimeModule = new TslRuntimeModule()
+		val webModule = new TslWebModule(executorServiceProvider)
+		
+		if( !EPackage.Registry.INSTANCE.containsKey("http://www.testeditor.org/tsl")){

```

Comment:
As stated above, the registration of the ePackage should go into `TslStandaloneSetup`, then this is no longer necessary here.